### PR TITLE
CARGO: fetch real stdlib metadata

### LINKED ISSUE #3835: Cannot find declaration of BTreeMap and BTreeSet
## Environment

* **IntelliJ-Rust plugin version: 0.2.98.2632-191-nightly**
* **Rust toolchain version: 1.34.0 (91856ed52 2019-04-10)**
* **IDE name and version: CLion 2019.1.3**
* **Operating system: macOS 10.14.4**

## Problem description

```rust
use std::collections::{BTreeMap, BTreeSet};
//                     ^         ^
//                   'Cannot find declaration to go to'
```

## Steps to reproduce

Type `use std::collections::{BTreeMap, BTreeSet}` and click `BTreeMap` and `BTreeSet` with cmd/ctrl key down.

### PR DESCRIPTION
**Pull request by Undin**
It allows us to get actual info (like edition) about stdlib crates.
It's important because name resolution rules depend on edition

Fixes #3835

### CODE REVIEW

**Comment by Undin** [comment]
bors try

File: src/main/kotlin/org/rust/lang/core/psi/RsCodeFragmentFactory.kt [hunk 1]
```diff
@@ -20,10 +22,17 @@ class RsCodeFragmentFactory(val project: Project) {
     private val psiFactory = RsPsiFactory(project, markGenerated = false)
 
     fun createCrateRelativePath(pathText: String, target: CargoWorkspace.Target): RsPath? {
-        check(pathText.startsWith("::"))
+        require(!pathText.startsWith("::")) { "Path shouldn't start with `::`" }
+        require(!pathText.startsWith("crate::")) { "Path shouldn't start with `crate::`" }
+
+        val prefix = when (target.edition) {
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/psi/RsCodeFragmentFactory.kt]: Looks like this isn't needed

Comment (Undin) [reply src/main/kotlin/org/rust/lang/core/psi/RsCodeFragmentFactory.kt]: Do you think that path without `::` prefix is enough?

Comment (vlad20012) [reply src/main/kotlin/org/rust/lang/core/psi/RsCodeFragmentFactory.kt]: I do it this way in #3828 and looks like it works

**Comment by Undin** [comment]
bors try

**Comment by Undin** [comment]
bors try

**Comment by Undin** [comment]
bors try

**Pull request closed without merging**